packages/flutter/test/cupertino/route_test.dart

```diff
@@ -784,47 +784,56 @@ void main() {
     expect(tester.getTopLeft(find.byType(Placeholder)).dx, moreOrLessEquals(0.0, epsilon: 0.1));
     await tester.pump();
 
-    // We use a higher number of intervals since the animation has to scale the
-    // entire screen.
-
-    await tester.pump(const Duration(milliseconds: 40));
-    expect(tester.getTopLeft(find.byType(Placeholder)).dx, moreOrLessEquals(-55.0, epsilon: 1.0));
+    // The whole transition is 500ms based on
+    // CupertinoPageRoute.transitionDuration. Break it up into 10 small chunks.
+    //
+    // The screen width is 800.
+    // The top left corner of the placeholder will go from 0 to -800 / 3 = -266.67.
 
-    await tester.pump(const Duration(milliseconds: 40));
-    expect(tester.getTopLeft(find.byType(Placeholder)).dx, moreOrLessEquals(-111.0, epsilon: 1.0));
+    await tester.pump(const Duration(milliseconds: 50));
+    expect(tester.getTopLeft(find.byType(Placeholder)).dx, moreOrLessEquals(-69.0, epsilon: 1.0));
 
-    await tester.pump(const Duration(milliseconds: 40));
-    expect(tester.getTopLeft(find.byType(Placeholder)).dx, moreOrLessEquals(-161.0, epsilon: 1.0));
+    await tester.pump(const Duration(milliseconds: 50));
+    expect(tester.getTopLeft(find.byType(Placeholder)).dx, moreOrLessEquals(-136.0, epsilon: 1.0));
```

Comment:
I believe this test was outdated, the total duration of the page is 500ms, so it makes more sense to update with 10 chunks of 50ms instead of 40ms.

During the forward (push), the value _"didn't change"_ (meaning the test was passing). I only updated them because I changed the duration in the `pump()`s.

Reply:
Can you create a new test rather than changing the existing one? We want to preserve the original test to prevent regressions.